aeron-driver/src/main/java/io/aeron/driver/DriverManagedResource.java

```diff
@@ -18,7 +18,7 @@
 /**
  * Common behaviour supported for driver resources such as publications and images.
  */
-public interface DriverManagedResource extends AutoCloseable
+public interface DriverManagedResource
```

Comment:
Why remove `AutoCloseable`?

Reply:
Because I removed the free operation from the close within the various driver managed resources I wanted to discourage people from thinking that something like the following would be valid:
```
try (final NetworkPublication publication = ....)
{
    // Do stuff
}
```
This is not really valid now as it will leak file resources.  I would like the change the name of the `close` operation, but not sure what would be correct.

Reply:
One consequence of making `DriverManagedResource` not `AutoCloseable` is that the actual invocation of the `close` operation now allocates:
```java
CloseHelper.close(ctx.errorHandler(), resource::close);
```

Reply:
Yes, I'm conflicted on that one.  Allocation vs indicating that a language feature applies when it doesn't.  The other option would be to inline the behaviour from the CloseHelper.  I'll have a look at what feels best.

Reply:
If the intent is to remove the `AutoClosable` then inlining the `CloseHelper` logic seems the most straightforward to me.

Reply:
I ran the driver through the async-profiler's allocation profiler and it didn't show any allocations for that section of the code.  It is possible in some cases that a method reference can allocate, but in this particular case I can't see any evidence of it.  While adding and closing 20,000 exclusive publications, all of the allocations within the `DriverConductor::checkManagedResources` call tree come from `ReceiverProxy::removeCoolDown`, which has a capturing lambda.  I'm going to leave the code as is.

Reply:
Actually those lambdas do allocate and you can see it if you run with the `-XX:+UseG1GC -XX:-UseTLAB` options (which disables TLAB and allows async-profiler to see every allocation).
![Allocating_lambdas_when_closing](https://user-images.githubusercontent.com/696855/212361678-620fd4c4-3201-4152-a217-9003a0531029.png)
As you can see from the above the line [DriverConductor.java:1977](https://github.com/mikeb01/Aeron/blob/7469a86ae3423a8464a04d8e657cf29734c37f1b/aeron-driver/src/main/java/io/aeron/driver/DriverConductor.java#L1977) does allocate.

Reply:
Allocation here is not a big issue. The resources have to be allocated and don't churn that often.